contracts/libraries/concentratedPool/FullMath.sol

```diff
@@ -113,7 +118,7 @@ library FullMath {
         result = mulDiv(a, b, denominator);
         unchecked {
             if (mulmod(a, b, denominator) != 0) {
-                require(result < type(uint256).max);
+                if (result > type(uint256).max) revert ResultMax();
```

Comment:
This doesn't work. you can never go above max. it'll wrap around.

Can we continue using requires?

Reply:
that would be my preference @maxsam4 maybe we can treat this PR as something that could save space if we run into limits